### PR TITLE
add LazyFieldView to OperatorIO

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Add a new LazyFieldView to OperatorIO which only applies changes in the input field when a save button within an input field is clicked or optionally (enabled by default) when you blur the input field. When a field is unsaved, reset button (to go back to the previously applied or initial state) and save button (to apply changes) will appear

Preview (unsaved state of a LazyFieldView):
image

Usage:

```
def resolve_input(self, ctx):
    inputs = types.Object()
    inputs.str("url", view=types.LazyFieldView(save_on_blur=False))
    return types.Property(inputs)
```

## How is this patch tested? If it is not, please explain why.

Using a test operator in the app

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

```
- Added a new :class:`LazyFieldView <fiftyone.operators.types.LazyFieldView>` type that extends capabilities of :class:`FieldView <fiftyone.operators.types.FieldView >` to allow applying user changes in an input field only with a click of a save button within an input field; or, optionally (enabled by default) with a input field blur event on an input field `#3728<https://github.com/voxel51/fiftyone/pull/3728>`_
```

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: fiftyone/operators/types.py
Comment (brimoor) [inline fiftyone/operators/types.py:1391]: For Python docstrings, I believe Sphinx requires lines 2 and beyond of multiline parameter descriptions to be indented 4 spaces in order to be correctly rendered. For example:
https://github.com/voxel51/fiftyone/blob/48b050ba52de95dd7b8501d807485632f24ca172/fiftyone/core/dataset.py#L2864

File: fiftyone/operators/types.py
Comment (brimoor) [inline fiftyone/operators/types.py:1390]: nit: no strong preference but could consider `save_on_blur=False` to be more concise, with default being `save_on_blur=True`

**Review by brimoor — COMMENTED** [review]
I defer to code owners for review but LGTM! Just left a small comment

File: fiftyone/operators/types.py
Comment (swheaton) [reply fiftyone/operators/types.py:1390]: Just lurking unwanted here but @imanjra as a plug-in dev i have no idea what blur means in this context, would be helpful to explain in the docstring

**Review by ritch — APPROVED** [review]
LGTM. Please add an example to the fiftyone-plugins examples.

**Comment by imanjra** [comment]
@ritch Thank you for the review! Added an example operator https://github.com/voxel51/fiftyone-plugins/pull/89

**Pull request merged**
